python/lsst/ts/tcpip/one_client_read_loop_server.py

```diff
@@ -0,0 +1,143 @@
+        or `LOCALHOST_IPV6` for IPV6. If `None` then bind to all network
+        interfaces (e.g. listen on an IPv4 socket and an IPv6 socket).
+        None can cause trouble with port=0;
+        see ``port`` in the Attributes section for more information.
+    port : `int`
+        IP port for this server. If 0 then randomly pick an available port
+        (or ports, if listening on multiple sockets).
+        0 is strongly recommended for unit tests.
+    log : `logging.Logger`
+        Logger.
+    connect_callback : callable or `None`, optional
+        Asynchronous or (deprecated) synchronous function to call when
+        when a client connects or disconnects.
+        If the other end (client) closes the connection, it may take
+        ``monitor_connection_interval`` seconds or longer to notice.
+        The function receives one argument: this `OneClientServer`.
+    name : `str`, optional
+        Name used for log messages, e.g. "Commands" or "Telemetry".
+    **kwargs : `dict` [`str`, `typing.Any`]
+        Additional keyword arguments for `asyncio.start_server`,
+        beyond host and port.
+
+    Attributes
+    ----------
+    read_loop_task : `asyncio.Future`
+        A task that reads messages from the reader in a loop.
+    """
+
+    def __init__(
+        self,
+        host: str | None,
+        port: int | None,
+        log: logging.Logger,
+        connect_callback: ConnectCallbackType | None = None,
+        name: str = "",
+        **kwargs: typing.Any,
+    ) -> None:
+        self.read_loop_task: asyncio.Future = asyncio.Future()
+
+        if connect_callback is None:
+            connect_callback = self.connect_callback
+
+        super().__init__(
+            host=host,
+            port=port,
+            log=log,
+            connect_callback=connect_callback,
+            monitor_connection_interval=0,
+            name=name,
+            **kwargs,
+        )
+
+    async def connect_callback(self, server: BaseClientOrServer) -> None:
+        """A client has connected or disconnected."""
+        if self.connected:
+            self.log.info("Client connected.")
+            self.read_loop_task.cancel()
+            self.read_loop_task = asyncio.create_task(self.read_loop())
+        else:
+            self.log.info("Client disconnected.")
+
+    async def read_loop(self) -> None:
+        """Read incoming messages and handle them."""
+        try:
+            self.log.info(f"The read_loop begins connected? {self.connected}")
+            while self.connected:
+                self.log.debug("Waiting for next incoming message.")
+                await self.read_and_dispatch()
+        except Exception as e:
+            self.log.exception(f"read_loop failed. Disconnecting. {e!r}")
+            await self.disconnect(cancel_read_loop_task=False)
+
+    @abc.abstractmethod
+    async def read_and_dispatch(self) -> None:
+        """Read and dispatch.
```

Comment:
Can you expand this brief description just a bit? Perhaps something like "Read, parse, and dispatch one item of data". Or "one message" -- in which case perhaps "message" is better than "data". Sigh.

Reply:
I stuck with `data`.